proposals/data-transfer-proposal.md

```diff
@@ -0,0 +1,99 @@
+*   We understand the core technologies of Bitswap and Graphsync well enough to make big improvements in data transfer
+*   Our benchmarking suite is mature enough to provide real feedback
+
+#### User workflow example
+
+A dapp developer builds a centralized file transfer application. When one of their users adds a 1GiB file another user is able to download that 1GiB file reliably, with speeds close to an HTTP file download and with minimal overhead. Additionally, if 10 users all have the same 1 GiB file an 11th user should be able to download the data much faster than a single download over HTTP and with relatively low overhead.
+
+#### Impact
+
+🔥🔥
+
+*   If users are able to reliably and quickly download files over IPFS then they will increase their development velocity by not worrying about IPFS data transfer issues, and will not be pushed away from IPFS.
+*   Additionally, for users who end up relying on gateways to find and transfer data having reliable and fast data transfer that experience will work seamlessly and keep them building on our ecosystem
+*   Success could result in freeing up time for infrastructure engineers running major IPFS gateways, as well as decreasing developer attrition and the number of bug reports on data transfer in go-ipfs.
+
+#### Leverage
+
+🎯🎯 
+
+What we learn here will be critical to delivering an awesome storage and retrieval experience in filecoin
+
+#### Confidence
+
+Ability to deliver the feature - 7 or 8 -- we have strong benchmarking tools and early evidence we can make big improvements to data transfer
+
+Confidence in impact - 2 - We have lots of anecdotal reports that transfer speeds continue to impact use of IPFS
+
+## Project definition
+#### Brief plan of attack
+
+Much of the groundwork for this project is already laid down. As stated, we have a strong benchmarking suite, and many good approaches to delivering speed improvements.
+
+Our first step is to build prototypes using a raw libp2p-bitswap-graphsync stack -- this will involve some modifications for both bitswap and graphsync. We will run out prototypes against our benchmarking suite and iterate till we have a strong solution.
+
+Once we have a strong solution, we will move to implementing in releasable form in IPFS - which will have to happen in tandem with IPFS in IPLD project, as we really can’t mix Bitswap and Graphsync in the current stack of services without integrating ipld-prime into the current dag service.
+
+#### What does done look like?
+
+*   We release a version of go-ipfs with significant data transfer speed improvements… unless....
+*   We find our prototyping does not lead to any clear path to improved performance and quit early
+
+#### What does success look like?
+
+Success means no one leaves the IPFS ecosystem because data transfer is too slow.
+
+Success means new people come into the IPFS ecosystem seeking a way to move data around quickly
```

Comment:
Having a super high value collab that would commit to pulling in IPFS with this improvement could make this compelling. Ex, if this is the gap for full production Netflix deployment that would be a compelling success case (but if this isn’t the blocker, that also says something about impact of landing these improvements successfully)